lib/compress/zstdmt_compress.c

```diff
@@ -1334,9 +1335,32 @@ size_t ZSTDMT_initCStream_internal(
     mtctx->allJobsCompleted = 0;
     mtctx->consumed = 0;
     mtctx->produced = 0;
+
+    /* update dictionary */
+    ZSTD_freeCDict(mtctx->cdictLocal);
+    mtctx->cdictLocal = NULL;
+    mtctx->cdict = NULL;
+    if (dict) {
+        if (dictContentType == ZSTD_dct_rawContent) {
+            mtctx->inBuff.prefix.start = (const BYTE*)dict;
+            mtctx->inBuff.prefix.size = dictSize;
+        } else {
+            /* note : a loadPrefix becomes an internal CDict */
+            mtctx->cdictLocal = ZSTD_createCDict_advanced(dict, dictSize,
+                                                        dictLoadMethod, dictContentType,
```

Comment:
We don't actually care about the `dictLoadMethod` here, right? If it was `ZSTD_dlm_byCopy`, the outer `ZSTD_CCtx` must have already token ownership of the buffer. So we can use `byRef` here.
```suggestion
                                                        ZSTD_dlm_byRef, dictContentType,
```